libs/hdf-converters/src/ckl-mapper/checklist-mapper.ts

```diff
@@ -0,0 +1,351 @@
+import {ExecJSON} from 'inspecjs';
+import _ from 'lodash';
+import {version as HeimdallToolsVersion} from '../../package.json';
+import {ChecklistJSONIX} from '../../types/checklistJsonix';
+import {
+  BaseConverter,
+  generateHash,
+  ILookupPath,
+  MappedTransform,
+  parseXmlToJsonix
+} from '../base-converter';
+import {CciNistMapping} from '../mappings/CciNistMapping';
+import {DEFAULT_STATIC_CODE_ANALYSIS_NIST_TAGS} from '../utils/global';
+import {
+  ChecklistObject,
+  ChecklistVuln,
+  createChecklistObject
+} from './checklistConstructor';
+import * as checklistMapping from './jsonixMapping';
+
+/**
+ * Tranformer function that splits a string and return array
+ * @param input - string of CCI references
+ * @returns ref - array of CCI references
+ */
+function cciRef(input: string): string[] {
+  return input.split('; ');
+}
+
+const CCI_NIST_MAPPING = new CciNistMapping();
+
+/**
+ * Transformer function that splits string and maps resulting array
+ * into NIST control tags
+ * @param input - string of CCI references
+ * @returns tag - array of NIST Control Tags
+ */
+function nistTag(input: string): string[] {
+  const identifiers: string[] = cciRef(input);
+  return CCI_NIST_MAPPING.nistFilter(
+    identifiers,
+    DEFAULT_STATIC_CODE_ANALYSIS_NIST_TAGS
+  );
+}
+
+enum ImpactMapping {
+  high = 0.7,
+  medium = 0.5,
+  low = 0.3
+}
+
+/**
+ * Transformer function that checks if the status is 'Not Applicable' returning a 0.
+ * Otherwise, maps severity to ImpactMapping
+ * @param vuln - checklist vulnerability object
+ * @returns impact - number 0.3, 0.5, or 0.7
+ */
+function transformImpact(vuln: ChecklistVuln): number {
+  if (vuln.status === 'Not Applicable') return 0.0;
+  const severity = findSeverity(vuln);
+  // check if severity does not exists within ImpactMapping throw new Error
+  //throw new Error("Mapping does not exist");
+  return ImpactMapping[
+    severity.toLowerCase() as keyof typeof ImpactMapping
```

Comment:
find out what happens when the assumption that the string will be only low/medium/high is broken.  does it crash the program (that's fine i guess), can we catch the error and format it more nicely before throwing it (better behavior), what happens if the value is missing/null?

Reply:
StigViewer does not 'break' with a value for severity different than low/medium/high, however, it does not display a Severity at all, and in some cases, it will pop the Security Override Justification each time the control is selected. I am thinking this is a bug versus an expected behavior. This should allow the schema to remain as is, but add rigidity to the low/medium/high requirement when loading.